pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -353,6 +350,7 @@ def __init__(  # pylint: disable=too-many-arguments
 
         # Creating the state vector
         self._mpi_handler = MPIHandler(mpi, mpi_buf_size, len(self.wires), c_dtype)
+        self._use_mpi = mpi
```

Comment:
Shouldn't we use mpi from the MPI Handler?

Reply:
It doesn't make too much difference to me since this private data variable is going to be removed once mpi support for mid-measurement PR (#944) is in.

Reply:
I'm not convinced yet! This is part of the current design to use the protected mpi handler in LGPU device implementation. We should still adhere to the current design even though this is not following good practices. Revisiting the API can be done in a follow-up PR for the entire use cases of the mpi handler then.

Reply:
Let's add a use_mpi method to the mpi_handler class